### DISCUSSION TITLE
Re-export all public api dependencies

### DISCUSSION THREAD

**Discussion by tylerhawkes** [Byron/google-apis-rs#276] [Ideas]
A lot of the examples show that to use `google-storage1` or any other `google-*` crate that you need to include the crate and the correct version of hyper and some other things in `Cargo.toml`. Would you consider doing something like 
```
pub mod deps {
    pub use yup_oauth2;
}
```
so that consumers of the api could do 
```
use google_storage1::deps::*;
```
and then we wouldn't have to exactly match hyper versions or have clients depend on crates that are part of the public api to be able to use them?

**Comment by Byron** [comment]
[Work on the next version is currently underway](https://github.com/google-apis-rs/generator), and we will absolutely consider improving on this, while we are at it.

Is this a blocking issue for you?

CC @ggriffiniii
